src/chapters/shell-intro/walkthrough_bash-basics.rst

```diff
@@ -0,0 +1,1043 @@
+
+Executing a script
+^^^^^^^^^^^^^^^^^^
+
+The act of executing a script is the same evaluating a command entered into the REPL. An interpreter is needed to understand and execute the contents. In the REPL the interpreter is implied by the Shell -- BASH in our case. The Shell is actually a BASH REPL process that is listening for, interpreting and executing commands.
+
+Recall that files in Linux are just strings of characters. It is **up to the program that interprets it** to decide what to do with its contents. When executing a script we have to define what program, like BASH or Python, that will interpret it. This can be done explicitly or implicitly. 
+
+Explicit execution is when we use the interpreter program (``bash``) as a command and provide it with the path to the script file as an argument. For example if we had a BASH script file in our home (``~``) directory we can execute it like this:
+
+.. sourcecode:: bash
+   :caption: Linux/BASH
+
+   $ bash ~/my-script.sh
+
+In other words when we want to execute a script explicitly we (the user) define the program we want to interpret it -- one of the core tenants covered earlier. As another example if we had a Python script we would naturally use the ``python`` interpreter to execute it:
+
+.. sourcecode:: bash
+   :caption: Linux/BASH
+
+   $ python ~/my-python-script.py
+
+If we try to execute a script with the wrong interpreter it will, expectedly, fail. It would be like handing an English-speaking person a book in Japanese and asking them to interpret it!
+
+.. admonition:: note
+
+   Implicit execution is when the interpreter is defined inside the script using a special line at the top of the script file called a `shebang <https://linuxize.com/post/bash-shebang/>`_. 
+   
+   Implicit execution is more advanced but is the standard approach when working with scripts professionally. We will not be covering implicit execution at this time as it involves some other steps that are best covered when you are more comfortable with scripting and BASH in general.
+
+The Executing User
+^^^^^^^^^^^^^^^^^^
+
+Commands in scripts are executed by the *user who executes the script*. While this might seem intuitive it does have an important implication. 
+
+This means that if you (``student``) run a script then all of those commands will be issued by the ``student`` user and be *subject to the permissions of that user*. If you need to run privileged commands you must run it using a super user account. 
+
+However, if you try to use ``sudo`` in the script a prompt would require you to authenticate as soon as it is executed -- a manual step that would defeat the purpose! Fortunately in the cloud the scripts we execute will run as ``root`` automatically and will not require the use of ``sudo``. 
```

Comment:
Move "Fortunately in the cloud" to a fun fact. to separate how cloud servers work, and how their local machine behaves.